Training/Setup/Patch/Data/AddCustomerAttributeCreditHold.php

```diff
@@ -0,0 +1,154 @@
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+namespace Codifi\Training\Setup\Patch\Data;
+
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Setup\Patch\PatchRevertableInterface;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Customer\Setup\CustomerSetupFactory;
+use Magento\Customer\Setup\CustomerSetup;
+use Magento\Customer\Model\ResourceModel\Attribute as AttributeResourceModel;
+use Magento\Eav\Model\Entity\Attribute\SetFactory as AttributeSetFactory;
+
+/**
+ * Class AddCustomerAttributeCreditHold
+ * @package Codifi\Training\Setup\Patch\Data
+ */
+class AddCustomerAttributeCreditHold implements DataPatchInterface, PatchRevertableInterface
+{
+
+    const CODE_CUSTOM_ATTRIBUTE = 'credit_hold';
+
+    const LABEL_CUSTOM_ATTRIBUTE = 'Credit Hold';
+
+    /**
+     * @var ModuleDataSetupInterface
+     * An additional pre-configuration environment that creates a database connection
+     */
+    private $moduleDataSetup;
+
+    /**
+     * @var CustomerSetupFactory
+     */
+    private $customerSetupFactory;
+
+    /**
+     * @var AttributeSetFactory
+     */
+    private $attributeSetFactory;
+
+    /**
+     * @var AttributeResourceModel
+     */
+    private $attributeResourceModel;
+
+    /**
+     * AddCustomerAttributeCereditHold Constructor
+     *
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param CustomerSetupFactory $customerSetupFactory
+     * @param AttributeSetFactory $attributeSetFactory
+     * @param AttributeResourceModel $attributeResourceModel
+     */
+    public function __construct(
+        ModuleDataSetupInterface $moduleDataSetup,
+        CustomerSetupFactory $customerSetupFactory,
+        AttributeSetFactory $attributeSetFactory,
+        AttributeResourceModel $attributeResourceModel
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->customerSetupFactory = $customerSetupFactory;
+        $this->attributeSetFactory = $attributeSetFactory;
+        $this->attributeResourceModel = $attributeResourceModel;
+    }
+
+    /**
+     * Add 'Credit Hold' customer attribute
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->getConnection()->startSetup();
+        /** @var CustomerSetup $customerSetup */
+        $customerSetup = $this->customerSetupFactory->create(['setup' => $this->moduleDataSetup]);
+
+        $customerEntity = $customerSetup->getEavConfig()->getEntityType(\Magento\Customer\Model\Customer::ENTITY);
+        $attributeSetId = $customerEntity->getDefaultAttributeSetId();
+
+        $attributeSet = $this->attributeSetFactory->create();
+        $attributeGroupId = $attributeSet->getDefaultGroupId($attributeSetId);
+
+        $customerSetup->addAttribute(
+            \Magento\Customer\Model\Customer::ENTITY,
+            self::CODE_CUSTOM_ATTRIBUTE,
+            [
+                'attribute_set_id' => '',
```

Comment:
'attribute_set_id' and 'attribute_group_id' are redundant in this data array as you add them in next step